src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -511,19 +546,8 @@ struct MinimumSpanningTree
         Kokkos::view_alloc(Kokkos::WithoutInitializing, "ArborX::MST::radii"),
         n);
 
-    bool const use_lower_bounds =
-#if defined(KOKKOS_ENABLE_SERIAL)
-        std::is_same<ExecutionSpace, Kokkos::Serial>::value;
-#else
-        false;
-#endif
     Kokkos::View<float *, MemorySpace> lower_bounds("ArborX::MST::lower_bounds",
-                                                    0);
-    if (use_lower_bounds)
-    {
-      reallocWithoutInitializing(lower_bounds, n);
-      Kokkos::deep_copy(space, lower_bounds, 0);
-    }
+                                                    n);
```

Comment:
Why unconditional allocation and init?

Reply:
OK I see we compute the `lower_bounds` all the time now.  Add a comment that we can revisit that when we have C++17